pallets/asset/src/lib.rs

```diff
@@ -1207,6 +1208,11 @@ impl<T: Config> Module<T> {
             // Everything after must be a NULL byte.
             .map_or(true, |nm_pos| bytes[nm_pos..].iter().all(|b| *b == 0));
         ensure!(good, Error::<T>::TickerNotAscii);
+
+        for ref_b in bytes {
+            let deref_b = *ref_b;
+            ensure!(deref_b.is_ascii_alphanumeric(), Error::<T>::TickerNotAscii);
+        }
```

Comment:
This check will not allow trailing NULL bytes.

This code on line 1206, needs to be changed to use the `is_ascii_alphanumeric` call.
```
.position(|b| !matches!(b, 32..=126))
```